task4/VCS/Logger.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text.Json;
+
+namespace VCS
+{
+    /// <summary>
+    /// This class helps to save to the json file and get from it all commits and changes
+    /// </summary>
+    public class Logger
+    {
+        public List<Commit> Commits { get; private set; }
+        public List<FileEntity> InitialFileList { get; private set; }
+
+        public Logger()
+        {
+            Commits = new List<Commit>();
+            InitialFileList = new List<FileEntity>();
+        }
+
+        #region Save Origin Structure of directory to watch 
+        public void SaveStartStateToJson()
+        {
+            if (!File.Exists(Storage.InitialLog))
+            {
+                var files = Directory.GetFiles(Storage.DirectoryToWatch, "*.txt", SearchOption.AllDirectories);
+                foreach (var file in files)
+                {
+                    InitialFileList.Add(new FileEntity(file, File.ReadAllText(file)));
+                }
+                var json = JsonSerializer.Serialize(InitialFileList, new JsonSerializerOptions { WriteIndented = true });
+                if (string.IsNullOrWhiteSpace(json))
+                {
+                    File.WriteAllText(Storage.InitialLog, "{}");
+                }
+                File.WriteAllText(Storage.InitialLog, json);
```

Comment:
@STeeL835, what is the problem with saving null values or an empty collection? If the null file is empty or its content is not important. If the collection is empty, it means that the file directory is empty and it can be.

Reply:
> what is the problem with saving null values or an empty collection? 

No problem. I was explaining why you don't need to check string after serialization.

Reply:
ah, okay. I just didn't understand you